snap/snapcraft.yaml

```diff
@@ -1,56 +1,83 @@
 name: matter-pi-gpio-commander
-version: "0.1"
-summary: Matter Raspberry Pi GPIO output commander
-description: |
-  TBA
-license: Apache-2.0
-confinement: strict
-base: core22
+version: "0.2"
+summary: Raspberry Pi GPIO as a Matter lighting app
+description: Refer to https://snapcraft.io/matter-pi-gpio-commander
+
 grade: devel
+confinement: strict
 
+base: core22
 architectures:
   - build-on: arm64
   - build-on: armhf
+  # - build-on: amd64
 
 layout:
   /mnt:
-    bind: $SNAP_DATA/mnt
+    bind: $SNAP_COMMON/mnt
 
 parts:
-  build-res:
-    plugin: dump
-    source: build
+  wiring-pi:
+    plugin: nil
+    source: https://github.com/WiringPi/WiringPi.git
+    source-depth: 1
+    source-tag: 2.61-1
     override-build: |
-      cp requirements.txt $CRAFT_STAGE/
+      WIRINGPI_SUDO="" ./build
+
+      mkdir -p $CRAFT_PART_INSTALL/lib
+      cp -v /usr/local/lib/libwiringPi.so* $CRAFT_PART_INSTALL/lib/
+    build-packages: [git, gcc, g++]
+
+  test-blink:
+    after: [wiring-pi]
+    plugin: nil
+    source: .
+    override-build: |
+      g++ -Wall -o test-blink test-blink.cpp -lwiringPi
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp test-blink $CRAFT_PART_INSTALL/bin/
+  
   connectedhomeip:
-    after: [build-res]
     plugin: nil
-    override-build: |   
-      git clone https://github.com/project-chip/connectedhomeip.git --depth=1 .
+    source: https://github.com/project-chip/connectedhomeip.git
+    source-depth: 1
+    source-tag: v1.0.0.2
+    source-submodules: []
+    override-pull: |
+      craftctl default
+      # shallow clone the submodules
       scripts/checkout_submodules.py --shallow --platform linux
 
-      # The project writes its data to /tmp. Snap does not allow bind mounting
-      #   a persistent directory on /tmp, so we need to replace it in the code.
-      # Replace all /tmp directories with /mnt:
+  lighting:
+    after: [wiring-pi, connectedhomeip]
+    plugin: nil
+    source: app
+    override-build: |
+      cd ../../connectedhomeip/src
+
+      # Replace storage paths:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h
+      # Replace key-value store path:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h
```

Comment:
could you keep the explanation of the reason for replacing all /tmp with /mnt?

"The project writes its data to /tmp. Snap does not allow bind mounting
a persistent directory on /tmp, so we need to replace it in the code.
Replace all /tmp directories with /mnt:"

Reply:
Must have been removed by mistake. I'll add it back.

One may ask why we didn't just replace `/tmp` with `SNAP_DATA/some-path` and skip the layout mapping. That is because `SNAP_DATA` has the revision number inside which we don't know during build time. We added `/mnt` and mounted that on `SNAP_DATA`, dynamically resolving to the correct path regardless of the revision.

Later on, we changed to use `SNAP_COMMON` which actually doesn't have the revision inside. We have to take another look into that and check if setting the path directory during build is better. It will show up in logs and it is easier to debug. But it may break if the snap is installed with a different instance key (parallel installs).

Moreover, since this is a desktop app, it is more appropriate to keep the files in user directories (e.g. `SNAP_USER_COMMON`) which also allows using the CLI without privileges.